Examples/Swift/ManageOfflineRegionsExample.swift

```diff
@@ -0,0 +1,220 @@
+            pack!.resume()
+
+        }
+    }
+
+    // MARK: - MGLOfflinePack notification handlers
+
+    @objc func offlinePackProgressDidChange(notification: NSNotification) {
+        /**
+         Get the offline pack this notification is referring to,
+         along with its associated metadata.
+         */
+        if let pack = notification.object as? MGLOfflinePack,
+           let userInfo = NSKeyedUnarchiver.unarchiveObject(with: pack.context) as? [String: String] {
+
+            // At this point, the offline pack has finished downloading.
+            if pack.progress.countOfResourcesCompleted == pack.progress.countOfResourcesExpected {
+
+                let byteCount = ByteCountFormatter.string(fromByteCount: Int64(pack.progress.countOfBytesCompleted), countStyle: ByteCountFormatter.CountStyle.memory)
+
+                let packName = userInfo["name"] ?? "unknown"
+
+                print("""
+                  Offline pack “\(packName)” completed download:
+                    - Bytes: \(byteCount)
+                    - Resource count: \(pack.progress.countOfResourcesCompleted)")
+                """)
+
+            }
+        }
+        // Reload the table to update the progress percentage for each offline pack.
+        self.tableView.reloadData()
+
+    }
+
+}
+
+fileprivate extension MGLOfflinePackProgress {
+    var percentCompleted: Float {
+        let percentage = Float((countOfResourcesCompleted / countOfResourcesExpected) * 100)
+        return percentage
+    }
+}
+
+extension ManageOfflineRegionsExample_Swift: UITableViewDelegate, UITableViewDataSource {
+
+    // Create the table view which will display the downloaded regions.
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+
+        if let packs = MGLOfflineStorage.shared.packs {
+            return packs.count
+        } else {
+            return 0
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let label = UILabel()
+
+        label.backgroundColor = UIColor.systemBlue
+        label.textColor = UIColor.white
+        label.font = UIFont.preferredFont(forTextStyle: .headline)
+        label.textAlignment = .center
+
+        if MGLOfflineStorage.shared.packs != nil {
+            label.text = "Offline maps"
+        } else {
+            label.text = "No offline maps"
+        }
+
+        return label
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 50.0
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+
+        let cell = UITableViewCell.init(style: .subtitle, reuseIdentifier: "cell")
+
+        if let packs = MGLOfflineStorage.shared.packs {
+            let pack = packs[indexPath.row]
+
+            cell.textLabel?.text = "Region \(indexPath.row + 1): size: \(pack.progress.countOfBytesCompleted)"
```

Comment:
I think we should convert `pack.progress.countOfBytesCompleted` to MB, such that 600000 bytes could be transformed to a string such as `0.6 MB`.

Check out [`ByteCountFormatter`](https://developer.apple.com/documentation/foundation/bytecountformatter) which helps you do this. You'll want to create the `ByteCountFormatter` _outside_ of this method (so in a global constant) so it only gets created once for the entire table instead of for a per-cell basis, and then call the appropriate method to format the value here.

If you don't end up implementing this, I'd at least add the units to the end of this string as the user doesn't know what unit value is being displayed.